src/domain/profile/profile.service.ts

```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@nestjs/common';
+import { ProfileCoreRepository } from 'src/infrastructure/profile/profile.core-repository';
+import { ProfileResponseEntity } from 'src/domain/profile/entity/profile.response.entity';
+import { DomainCustomException } from 'src/domain/common/errors/domain-custom-exception';
+import { DomainErrorCode } from 'src/domain/common/errors/domain-error-code';
+
+@Injectable()
+export class ProfileService {
+  constructor(private readonly profileCoreRepository: ProfileCoreRepository) {}
```

Comment:
```suggestion
export class ProfileService {
  constructor(private readonly profileRepository: ProfileRepository) {}
```
여기서는 인터페이스에 의존해야 DIP를 지킬 수 있지 않을까요?

제가 처음에 설명드린것처럼, 제가 지금 만들어 놓은 구조는 레이어드 아키텍처 + 클린 아키텍처 사상을 반영한 구조입니다.
그래서 interfaces -> application -> domain <- infrastructure 이렇게 의존성 방향을 가져가려고 합니다.
핵심은 가장 **고수준 모듈은 domain 레이어**을 지키는게 이런 구조를 가져간 핵심입니다.
그런데 이렇게 바로 구체 클래스를 의존해버리면,
interfaces -> application -> domain -> infrastructure
이렇게 되고, 저희가 지키고자 하는 domain 레이어를 지키지 못하게 됩니다.
혹시 다른 도메인은 어떻게 구성되어 있었는지 보셨을까요??